src/App/index.test.js

```diff
@@ -1,8 +1,51 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './index';
+import TestUtils from 'react-addons-test-utils';
 
-it('renders without crashing', () => {
-	const div = document.createElement('div');
-	ReactDOM.render(<App />, div);
+
+describe('App', () => {
+	const app = TestUtils.renderIntoDocument(<App/>);
+	it('renders without crashing', () => {
+		const div = document.createElement('div');
+		ReactDOM.render(<App />, div);
+	});
+	it('changeEndpoint sets the endpoint and loading to true, and clears data and errors', () => {
+		const mockEndpoint = "http://localhost:3002/alternateImages.json";
+		app.changeEndpoint(mockEndpoint);
+		const {endpoint, loading, data, error, subscription} = app.state;
+		expect(endpoint).toBe(mockEndpoint);
+		expect(loading).toBe(true);
+		expect(data).toBe(null);
```

Comment:
When you trigger change in endpoint we clear the data to prepare for the new data